src/shared-styles/_decoration.scss

```diff
@@ -0,0 +1,18 @@
+@use "color";
+@use "error";
+@use "grid";
+@use "logical";
+
+@mixin border($dimension: null) {
+
+  @if index((inline, inline-start, inline-end, block, block-start, block-end), $dimension) {
```

Comment:
Could be simplified to put most of the work on `logical.property`. Should be a followup i.e. not part of this PR.